content/developer/contributing/coding-standard-typescript.md

```diff
@@ -0,0 +1,662 @@
+
+Anonymous functions that are not an arrow function _MUST NOT_ be used.
+
+```ts
+// Good
+function foo(event: ClickEvent) {}
+
+const foo = function foo(event: ClickEvent) {};
+
+const foo = (event: ClickEvent) => {};
+
+document.addEventListener("click", (event: ClickEvent) => {})
+
+document.addEventListener("click", foo);
+
+// Bad
+const foo = function() {};
+
+document.addEventListener("click", function(event: ClickEvent) {})
+```
+
+### 6.4. Calling Functions
+
+When making a method or function call, there _MUST NOT_ be a space between the
+method or function name and the opening parenthesis, there _MUST NOT_ be a space
+after the opening parenthesis, and there _MUST NOT_ be a space before the
+closing parenthesis. In the argument list, there _MUST NOT_ be a space before
+each comma, and there _MUST_ be one space after each comma.
+
+```ts
+bar();
+foo.bar($arg1);
+Foo.baz($arg2, $arg3);
+```
+
+Argument lists _MAY_ be split across multiple lines, where each subsequent line
+is indented once. When doing so, the first item in the list _MUST_ be on the
+next line, and there _MUST_ be only one argument per line.
+
+```ts
+foo.bar(
+    $longArgument,
+    $longerArgument,
+    $muchLongerArgument
+);
+```
+
+## 57 Control Structures
+
+The general style rules for control structures are as follows. Many are automatically enforced by Prettier.
+
+- There _MUST_ be one space after the control structure keyword
+- There _MUST NOT_ be a space after the opening parenthesis
+- There _MUST NOT_ be a space before the closing parenthesis
+- There _MUST_ be one space between the closing parenthesis and the opening
+  brace
+- The structure body _MUST_ be indented once
+- The closing brace _MUST_ be on the next line after the body
+
+The body of each structure _MUST_ be enclosed by braces. This standardizes how
+the structures look, and reduces the likelihood of introducing errors as new
+lines get added to the body.
+
+### 7.1. `if`, `elseif`, `else`
+
+An `if` structure looks like the following. Note the placement of parentheses,
+spaces, and braces; and that `else` and `elseif` are on the same line as the
+closing brace from the earlier body.
+
+```ts
+if (expr1) {
+    // if body
+} else if (expr2) {
+    // else if body
+} else {
+    // else body;
+}
+```
+
+The keyword `elseif` _SHOULD_ be used instead of `else if` so that all control
+keywords look like single words.
+
+If statements _MUST_ have opening and closing brackets and be split onto multiple lines. Single line if statements are prohibited.
+
+### 7.2. `switch`, `case`
+
+A `switch` structure looks like the following. Note the placement of
+parentheses, spaces, and braces. The `case` statement _MUST_ be indented once
+from `switch`, and the `break` keyword (or other terminating keyword) _MUST_ be
+indented at the same level as the `case` body. There _MUST_ be a comment such as
+`// no break` when fall-through is intentional in a non-empty `case` body.
+
+```ts
+switch (expr) {
```

Comment:
did you update the linter to accept this format for the switch statements? Before it wanted the "switch" on the same line as the "case"s.

Reply:
It should be yes.